processing/src/test/java/org/apache/druid/concurrent/ExecsTest.java

```diff
@@ -118,6 +122,38 @@ public void run()
     producer.shutdown();
   }
 
+  @Test
+  public void testSynchronousQueueSingleThreadedExecutor() throws Exception
+  {
+    // The implementation of Execs.newBlockingSingleThreaded() rejectedExecutionHandler should not add tasks when it's in shutDown state
+    // When capacity is 0, a SynchronousQueue is used and if a task is put in it in ShutDown state, it will forever stuck in WAITING state
+    // as executor will not take() the task to schedule it.
+    final ListeningExecutorService intermediateTempExecutor = MoreExecutors.listeningDecorator(
+        Execs.newBlockingSingleThreaded("[TASK_ID]-appenderator-abandon", 0)
+    );
+    Callable<Void> task = () -> {
+      try {
+        Thread.sleep(1000); // Simulate long-running task
+      }
+      catch (InterruptedException e) {
+        Thread.sleep(1500);
```

Comment:
why do we need sleep() under catch?

Reply:
If we remove catch (InterruptedException e) do we lose anything?

Reply:
Yes, even if we remove this, even then the thread get stuck as the main problem is Task being added to the queue, without it being taken to execute. I did this to simulate RUNNING thread still being blocked at the time of adding another task being added. But it is not required